### PR TITLE
fix: Packed sequence performance regression and missing pad_seq_to_mult parameter

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary
When `pad_seq_to_mult == 1` (no padding for CP), skip computing `cu_seqlens_unpadded` to avoid triggering the slower TE kernel path. Only compute and include `cu_seqlens_unpadded` when `pad_seq_to_mult > 1` (actual padding for THD CP).

## Changes
- Add `pad_seq_to_mult` parameter to `GPTSFTPackedDataset`
- Conditionally compute `cu_seqlens_unpadded` based on `pad_seq_to_mult`
- Update `get_packed_seq_params` to only set padded variants when unpadded is present

## Files Modified
- `src/megatron/bridge/data/datasets/sft.py`
- `src/megatron/bridge/training/utils/packed_seq_utils.py`

### CODE REVIEW

File: src/megatron/bridge/training/utils/packed_seq_utils.py [hunk 1]
```diff
@@ -56,14 +56,23 @@ def get_packed_seq_params(batch: dict[str, torch.Tensor]) -> PackedSeqParams:
...
+            qkv_format="thd",
+        )
+    else:
+        return PackedSeqParams(
+            cu_seqlens_q=cu_seqlens_padded,
+            cu_seqlens_kv=cu_seqlens_padded,
+            max_seqlen_q=max_seqlen,
+            max_seqlen_kv=max_seqlen,
+            qkv_format="thd",
+        )
```
Comment (cuichenx) [inline src/megatron/bridge/training/utils/packed_seq_utils.py:78]: nit: is this change necessary? as long as cu_seqlens_unpadded is None, passing None to PackedSeqParams should also be fine

File: src/megatron/bridge/data/datasets/sft.py [hunk 2]
```diff
@@ -98,6 +98,7 @@ def create_sft_dataset(
     get_attention_mask_from_fusion: bool = True,
     pack_metadata_file_path: Path = None,
     pad_cu_seqlens: bool = False,
+    pad_seq_to_mult: int = 1,
```
Comment (cuichenx) [inline src/megatron/bridge/data/datasets/sft.py:101]: should you update the caller of `create_sft_dataset` in `finetuning_datasets.py` as well?

File: src/megatron/bridge/training/utils/packed_seq_utils.py (hunk 1 above)
Comment (yaoyu-33) [reply src/megatron/bridge/training/utils/packed_seq_utils.py:78]: i want to make it more clear here since we encounter the slow issue. just to make more explicit.
